### PR TITLE
Allow NProgress to attach to HTML Element parent

### PR DESCRIPTION
**Pull request by zshall**
Modified `Settings.parent` to accept either strings or HTML Elements.

* If set to a `string`, NProgress uses `querySelector` like before to get the parent.
* If set to an `HTMLElement` NProgress attaches directly to this parent element.

This change was made so that you can programmatically attach NProgress to dynamically generated DOM elements without needing to assign an ID or some other way of identifying the parent for `querySelector` to be able to find it.

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,35 @@
+{
```
Comment (RamyRais) [inline .vscode/launch.json]: Hi @zshall thank you for your PR.
Can you please remove this file, as it is vscode configuration related to your local dev environment.

File: nprogress.js [hunk 2]
```diff
@@ -221,7 +221,28 @@
    */
 
   NProgress.getElement = function() {
-    return document.querySelector(Settings.parent + ' > .nprogress');
+    var parent = NProgress.getParent();
+    // if the parent contains nested NProgress instances, select just the direct child
+    if (parent) {
+      return Array.prototype.slice.call(parent.querySelectorAll('.nprogress')).find(p => p.parentElement === parent);
```
Comment (RamyRais) [inline nprogress.js]: I see that you used the method find of arrays which is not supported on internet explorer.
Sorry it is been long time but as far as i remember the library should work on all browsers. So i don't want to introduce breaking change. You can use .filter and take the first element or to be more efficient have a for loop.

**Review by RamyRais — CHANGES_REQUESTED** [review]
Thank you so much for you contribution.
I left some somments.

File: nprogress.js (hunk 2 above)
Comment (zshall) [reply nprogress.js]: Sounds good! Working on that now.

File: .vscode/launch.json (hunk 1 above)
Comment (zshall) [reply .vscode/launch.json]: Yup, removed.

File: nprogress.js (hunk 2 above)
Comment (zshall) [reply nprogress.js]: Made the change. Unit tests are passing, testing it in my app now.

Comment (zshall) [reply nprogress.js]: Confirmed working both in my app and in the tests.

**Review by RamyRais — APPROVED** [review]

**Pull request merged**